plugins/woocommerce/tests/e2e-pw/bin/woocommerce-cleanup.php

```diff
@@ -341,29 +337,26 @@ function () {
 			'wc-cleanup/v1',
 			'/reset',
 			array(
-				'methods'             => 'POST',
+				'methods'             => 'GET',
 				'callback'            => 'wc_cleanup_reset_site_via_api',
 				'permission_callback' => function () {
-					// Verify the nonce before processing the request.
-					if ( ! isset( $_POST['wc_cleanup_reset_nonce'] ) || ! wp_verify_nonce( sanitize_text_field( wp_unslash( $_POST['wc_cleanup_reset_nonce'] ) ), 'wc_cleanup_reset_action' ) ) {
-						return new WP_Error( 'rest_forbidden', esc_html__( 'Nonce verification failed', 'woocommerce' ), array( 'status' => 403 ) );
-					}
-
+					// phpcs:disable WordPress.Security.NonceVerification.Recommended
 					$provided_key = isset( $_GET['key'] ) ? sanitize_text_field( wp_unslash( $_GET['key'] ) ) : '';
-					$valid_key    = 'FUFP2UrAbJa_.GMfs*nXne*9Fq7abvYv'; // Replace with your actual secret key.
-					return $provided_key === $valid_key;
+						$valid_key    = 'FUFP2UrAbJa_.GMfs*nXne*9Fq7abvYv'; // Replace with your actual secret key.
```

Comment:
I just figured it was a small layer of protection against a crawler or something inadvertently triggering a reset.  There's no harm resetting the site, so I don't think we need heavy security, but it would be good if it wasn't in the middle of a test run.

Reply:
The API request should require authentication and we should use the site's credentials. 
All the routes in `test-helper-apis.php` are authenticated with basic auth, here's the latest addition: https://github.com/woocommerce/woocommerce/blob/84ccc69d7253a901368c1a7c56ffdfcad2eb095f/plugins/woocommerce/tests/e2e-pw/utils/themes.js#L7

Reply:
This worked for me as alternative to using key:

```
add_action(
	'rest_api_init',
	function () {
		register_rest_route(
			'wc-cleanup/v1',
			'/reset',
			array(
				'methods'             => 'GET',
				'callback'            => 'wc_cleanup_reset_site_via_api',
				'permission_callback' => function () {
					$auth_header = $_SERVER['HTTP_AUTHORIZATION'] ?? '';
					$credentials = base64_decode( substr( $auth_header, 6 ) );
					list( $username, $password ) = explode( ':', $credentials, 2 );
					$user = wp_authenticate( $username, $password );
					if ( is_wp_error( $user ) || ! user_can( $user, 'manage_woocommerce' ) ) {
						return new WP_Error( 'forbidden', 'Unauthorized', array( 'status' => 403 ) );
					}

					return true;
				},
			)
		);
	}
);
```

Reply:
Perfect, thanks @veljkho -- I did just that and updated global-setup to use the admin account for authentication.